server/src/info.ts

```diff
@@ -0,0 +1,72 @@
+import type { DocumentUri, Position } from 'vscode-languageserver';
+
+import type { TokenInfo, TokenLink, TokenLocation } from '@fluencelabs/aqua-language-server-api/aqua-lsp-api';
+
+import { locInLoc, posInToken } from './utils';
+
+/**
+ * Class that incapsulates information
+ * about tokens inside a document
+ */
+export class DocumentInfo {
+    private links: TokenLink[];
+    private tokens: TokenInfo[];
+
+    constructor(links: TokenLink[], tokens: TokenInfo[]) {
+        this.links = links;
+        this.tokens = tokens;
+    }
+
+    /**
+     * Find token info at position
+     * @param pos position to check
+     * @returns token info if found
+     */
+    infoAt(pos: Position): TokenInfo | undefined {
+        const info = this.tokens.find((token) => posInToken(pos, token.location));
```

Comment:
Yes, it could become a heavy computation. 
But this PR is mainly for fixing error thrown in `onHover` and refactoring.
This behavior is not changed, optimization could be done in separate PR.

Reply:
Agree. Just to remember the thing.